.github/workflows/continuous-integration.yaml

```diff
@@ -427,21 +427,81 @@ jobs:
           make actionlint
 
       - name: Source code conformance
-        id: cs
+        id: make-conformance
         if: ${{ steps.install-deps.outcome == 'success' && !cancelled() }}
         # pcc launches an internal target with proper flags
         run: |
           make pcc
 
+      # Checked for changes between main and the current branch in a PR. More specifically, 
+      # this is used in regular CIs to avoid launching Pytest, checking codeblocks, building docs 
+      # or other steps if the associated files were not touched. For most, we also check that the
+      # linux MD5 has not changed, which means that no libraries got updated. This is done in order
+      # to handle PRs which only upgrades dependencies
+      - name: Get all changed files from main in PR
+        id: changed-files-in-pr
+        if: | 
+          env.IS_PR
+          && !fromJSON(env.IS_WEEKLY) 
+          && !fromJSON(env.IS_RELEASE)
+          && steps.install-deps.outcome == 'success'
+          && steps.make-conformance.outcome == 'success' 
+          && !cancelled()
+        uses: tj-actions/changed-files@v37.5.1
+        with:
+          files_yaml: |
+            src:
+              - src/**
+            tests:
+              - tests/**
+              - src/concrete/ml/pytest/**
+              - conftest.py
+              - deps_licenses/licenses_linux_user.txt.md5
+            determinism:
+              - src/**
+              - tests/seeding/test_seeding.py
+              - conftest.py
+              - deps_licenses/licenses_linux_user.txt.md5
+            docs:
+              - docs/**
+            codeblocks:
+              - src/**
```

Comment:
so this was a bit trickier : codeblocks are found directly in the `script/make_utils/pytest_codeblocks.sh` script, using `MD_FILES=$(find . -type f -name "*.md" | grep -v "^\./\." | grep -v "api/concrete\.ml")`

my initial idea was maybe to have a previous step that does this grep in the CI, and then give it to the action + the `make pytest_codeblock` later, but I'm unsure that I'm able to give such a variable to the action like this, and thus I preferred to manually do it here : 
- `src`: because if source changed, codeblocks should be tested again
- `**.md`: because that's what we do in the above find / grep
- `!.*/**` and `!docs/_*/**` sort of the same as in the find, we don't want to consider md from hidden directories or stuf like `docs/_apidoc`
- `deps_licenses/licenses_linux_user.txt.md5` same as most, if some dependencies were changed, we need to test codeblocks

but I'll add a comment to make this more clear